src/Main/Mode/Batch.hs

```diff
@@ -122,95 +156,47 @@ run thisMode as
-          out (const Pretty.emptyDoc) (return . prettyOpenDiffTheory) (loadOpenDiffThy   as inFile)
-      | argExists "parseOnly" as || argExists "outModule" as =
-          out (const Pretty.emptyDoc) choosePretty (loadOpenThy as inFile)
-      | argExists "diff" as =
-          out ppWfAndSummaryDiff      (return . prettyClosedDiffTheory) (loadClosedDiffThy as inFile)
-      | otherwise        = do
-          (thy,report) <- loadClosedThyWf as inFile
-          out (ppWfAndSummary report) (return . prettyClosedTheory) (return thy)
+    processThy :: FilePath -> IO (Pretty.Doc, Pretty.Doc)
+    processThy inFile = either handleError return <=< runExceptT $ do
+      srcThy <- liftIO $ readFile inFile
+      thy    <- loadTheory thyLoadOptions srcThy inFile
+
+      if isParseOnlyMode then do
+        either (\t -> bisequence (liftIO $ choosePretty t, return Pretty.emptyDoc))
+               (\d -> return (prettyOpenDiffTheory d, Pretty.emptyDoc)) thy
+      else do
+        let sig = either (get thySignature) (get diffThySignature) thy
+        sig'   <- liftIO $ toSignatureWithMaude (get oMaudePath thyLoadOptions) sig
+
+        (report, thy') <- closeTheory thyLoadOptions sig' thy
+        either (\t -> return (prettyClosedTheory t,     ppWf report Pretty.$--$ prettyClosedSummary t))
+               (\d -> return (prettyClosedDiffTheory d, ppWf report Pretty.$--$ prettyClosedDiffSummary d)) thy'
       where
-        ppAnalyzed = Pretty.text $ "analyzed: " ++ inFile
-        ppWfAndSummary report thy = do
-            report
-            Pretty.$--$ prettyClosedSummary thy
-
-        ppWfAndSummaryDiff thy = do
-            reportWellformednessDoc $ checkWellformednessDiff (openDiffTheory thy) (get diffThySignature thy)
-            Pretty.$--$ prettyClosedDiffSummary thy
-
-        choosePretty = case getOutputModule as of
-          ModuleSpthy      -> return . prettyOpenTheory  <=< Sapic.warnings -- output as is, including SAPIC elements
-          ModuleSpthyTyped -> return . prettyOpenTheory <=< Sapic.typeTheory <=< Sapic.warnings  -- additionally type
-          ModuleMsr        -> return . prettyOpenTranslatedTheory
-            <=< (return . (filterLemma $ lemmaSelector as))
+        isParseOnlyMode = get oParseOnlyMode thyLoadOptions
+
+        handleError (ParserError e) = error $ show e
+        handleError (WarningError report) = do
+          putStrLn $ renderDoc $ Pretty.vcat [ Pretty.text ""
+                                             , Pretty.text "WARNING: the following wellformedness checks failed!"
+                                             , Pretty.text ""
+                                             , prettyWfErrorReport report
+                                             , Pretty.text "" ]
+          die "quit-on-warning mode selected - aborting on wellformedness errors."
+
+        -- FIXME: Does it make sense to print the warning even though not in prove mode?
```

Comment:
Like, if we only do `tamarin-prover $file`? Then I think the warning is not necessary.

Reply:
Fixed in 4ac36141f158f4da76a836aa4e901dd54c57c6fc